airflow/models/queue_task_run.py

```diff
@@ -0,0 +1,206 @@
+        iso = self.execution_date.isoformat()
+        cmd = ["airflow", "tasks", "run", str(self.dag_id), str(self.task_id), str(iso), "--raw"]
+        if self.mark_success:
+            cmd.extend(["--mark_success"])
+        if self.pickle_id:
+            cmd.extend(["--pickle", str(self.pickle_id)])
+        if self.job_id:
+            cmd.extend(["--job_id", str(self.job_id)])
+        if self.force:
+            cmd.extend(["--force"])
+        if self.pool:
+            cmd.extend(["--pool", self.pool])
+        if self.subdir:
+            cmd.extend(["--subdir", self.subdir])
+        if self.cfg_path:
+            cmd.extend(["--cfg_path", self.cfg_path])
+        return cmd
+
+    def __repr__(self):
+        iso = self.execution_date.isoformat()
+        return f"RawTaskDeferredRun(dag_id={self.dag_id}, task_id={self.task_id}, execution_date={iso})"
+
+
+class LocalTaskJobDeferredRun:  # pylint: disable=too-many-instance-attributes
+    """
+    Generates the shell command required to execute this task instance.
+
+    :param dag_id: DAG ID
+    :type dag_id: unicode
+    :param task_id: Task ID
+    :type task_id: unicode
+    :param execution_date: Execution date for the task
+    :type execution_date: datetime.datetime
+    :param mark_success: Whether to mark the task as successful
+    :type mark_success: bool
+    :param ignore_all_dependencies: Ignore all ignorable dependencies.
+        Overrides the other ignore_* parameters.
+    :type ignore_all_dependencies: bool
+    :param ignore_depends_on_past: Ignore depends_on_past parameter of DAGs
+        (e.g. for Backfills)
+    :type ignore_depends_on_past: bool
+    :param ignore_dependencies: Ignore task-specific dependencies such as depends_on_past
+        and trigger rule
+    :type ignore_dependencies: bool
+    :param force: Ignore the task instance's previous failure/success
+    :type force: bool
+    :param local: Whether to run the task locally
+    :type local: bool
+    :param pickle_id: If the DAG was serialized to the DB, the ID
+        associated with the pickled DAG
+    :type pickle_id: unicode
+    :param subdir: path to the file containing the DAG definition
+    :param raw: raw mode (needs more details)
+    :param job_id: job ID (needs more details)
+    :param pool: the Airflow pool that the task should run in
+    :type pool: unicode
+    :param cfg_path: the Path to the configuration file
+    :type cfg_path: str
+    :return: shell command that can be used to run the task instance
+    """
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        dag_id,
+        task_id,
+        execution_date,
+        mark_success=None,
+        pickle_id=None,
+        job_id=None,
+        ignore_all_dependencies=None,
+        ignore_dependencies=None,
+        ignore_depends_on_past=None,
+        force=None,
+        local=None,
+        pool=None,
+        raw=None,
+        subdir=None,
+        cfg_path=None,
```

Comment:
There's an _awful_ lot of overlap here with SimpleTaskInstance -- could this all instead by methods on SimpleTI (if we don't just remove this entirely as per my previous comment)

Reply:
They can't because these values are not known when creating SimpleTaskInstance. We can try to set the attribute later, but this is mixing concepts in my opinion.  But I prefer when the class has only one responsibility.